tools/tesseratos/include/tesseratos/toolbox/plugin.hpp

```diff
@@ -0,0 +1,67 @@
+/// @dir
+/// @brief @ref tesseratos-toolbox-plugin plugin directory.
+
+/// @file
+/// @brief Plugin entry point.
+/// @ingroup tesseratos-toolbox-plugin
+
+#pragma once
+
+#include <cubos/engine/cubos.hpp>
+
+namespace tesseratos
+{
+    /// @defgroup tesseratos-toolbox-plugin Toolbox
+    /// @ingroup tesseratos
+    /// @brief Adds a resource used to keep track of whether each tool is open or not.
+    /// ## Resources
+    /// - @ref Toolbox
+    ///
+    /// ## Dependencies
+    /// - @ref imguiPlugin
+
+    /// @brief Resource which manages other tools windows.
+    /// @ingroup tesseratos-toolbox-plugin
+    class Toolbox final
+    {
+    public:
+        /// @brief When called for the first time for a given name stores a new boolean for it, by default set to false.
+        /// @param toolName The tools name
+        ///@return Tools associated boolean.
+        bool isOpen(std::string toolName);
+
+        /// @brief Opens a tool
+        /// @param toolName The tools name
+        void open(std::string toolName);
+
+        /// @brief Closes a tool
+        /// @param toolName The tools name
+        void close(std::string toolName);
+
+        /// @brief Sets the boolean of a tool to its logical opposite state
+        /// @param toolName The tools name
+        void doClick(std::string toolName);
+
+        /// @brief Returns the beggining of the ToolsMap
+        /// @return begginng of the map
+        auto begin() const
+        {
+            return mToolsMap.begin();
+        }
+
+        /// @brief Returns the end of the ToolsMap
+        /// @return end of the map
+        auto end() const
+        {
+            return mToolsMap.end();
+        }
```

Comment:
```suggestion
        /// @brief Opens a tool.
        /// @param toolName Tool name.
        void open(const std::string& toolName);

        /// @brief Closes a tool.
        /// @param toolName Tool name.
        void close(const std::string& toolName);

        /// @brief If the given tool is open, closes it. Otherwise, opens it.
        /// @param toolName Tool name
        void toggle(const std::string& toolName);

        /// @brief Returns the beginning of the map with known tools.
        /// @return Beginning of the map.
        auto begin() const
        {
            return mToolsMap.begin();
        }

        /// @brief Returns the end of the map with known tools.
        /// @return End of the map.
        auto end() const
        {
            return mToolsMap.end();
        }
```